### ISSUE TITLE
Add AR versions to generators for migrations

### ISSUE THREAD

**Issue by bwlang** [QueueClassic/queue_classic#300]
In rails 6 migrations fail like this:

```
langhorst$ bin/rails generate queue_classic:install
      create  db/migrate/20190102140228_add_queue_classic.rb
      create  db/migrate/20190102140229_update_queue_classic_3_0_0.rb
      create  db/migrate/20190102140230_update_queue_classic_3_0_2.rb
      create  db/migrate/20190102140231_update_queue_classic_3_1_0.rb
 langhorst$ bundle exec rake db:migrate
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:

  class AddQueueClassic < ActiveRecord::Migration[4.2]
/Users/langhorst/src/depletion-design/db/migrate/20190102140228_add_queue_classic.rb:1:in `<main>'
/Users/langhorst/.rbenv/versions/2.5.3/bin/bundle:23:in `load'
/Users/langhorst/.rbenv/versions/2.5.3/bin/bundle:23:in `<main>'
```

This can be trivially fixed by adding a [4.2] to the end of each inheritance statement in the generated migrations, but it would be nice to fix the generators too.

I'm not sure how far backwards compatible this is, but I think at least to rails 4.

**Comment by ramfjord** [comment]
Also on rails 5.2:

```
root@paws-development-testing-0:/app# bundle exec rails generate queue_classic:install
Running via Spring preloader in process 441
      create  db/migrate/20190501014341_add_queue_classic.rb
      create  db/migrate/20190501014342_update_queue_classic_3_0_0.rb
      create  db/migrate/20190501014343_update_queue_classic_3_0_2.rb
      create  db/migrate/20190501014344_update_queue_classic_3_1_0.rb
root@paws-development-testing-0:/app# ^C
root@paws-development-testing-0:/app# rake db:migrate
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:

  class AddQueueClassic < ActiveRecord::Migration[4.2]
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:528:in `inherited'
/app/db/migrate/20190501014341_add_queue_classic.rb:1:in `<main>'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `block in require'
/usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
/usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:986:in `load_migration'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:982:in `migration'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:977:in `disable_ddl_transaction'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1350:in `use_transaction?'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1342:in `ddl_transaction'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1262:in `each'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1210:in `block in migrate'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1210:in `migrate'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1036:in `up'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1011:in `migrate'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/tasks/database_tasks.rb:172:in `migrate'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/railties/databases.rake:60:in `block (2 levels) in <main>'
/usr/local/bundle/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'

Caused by:
StandardError: Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:

  class AddQueueClassic < ActiveRecord::Migration[4.2]
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:528:in `inherited'
/app/db/migrate/20190501014341_add_queue_classic.rb:1:in `<main>'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/usr/local/bundle/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `block in require'
/usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
/usr/local/bundle/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:986:in `load_migration'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:982:in `migration'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:977:in `disable_ddl_transaction'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1350:in `use_transaction?'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1342:in `ddl_transaction'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1262:in `each'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1210:in `block in migrate'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1210:in `migrate'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1036:in `up'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/migration.rb:1011:in `migrate'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/tasks/database_tasks.rb:172:in `migrate'
/usr/local/bundle/gems/activerecord-5.2.0/lib/active_record/railties/databases.rake:60:in `block (2 levels) in <main>'
/usr/local/bundle/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
root@paws-development-testing-0:/app# rails --version
Rails 5.2.0
root@paws-development-testing-0:/app# grep queue_classic Gemfile.lock
  remote: https://github.com/QueueClassic/queue_classic.git
    queue_classic (3.2.0.RC1)
  queue_classic!
```

**Comment by ukd1** [comment]
@bwlang @ramfjord added fixes and tests for this in #308
